clients/admin-ui/src/features/common/hooks/dirty-forms.slice.ts

```diff
@@ -0,0 +1,91 @@
+import { createSelector, createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+import type { RootState } from "~/app/store";
+
+type FormState = {
+  id: string;
+  name: string;
+  isDirty: boolean;
+};
+
+type DirtyFormStates = {
+  forms: FormState[];
+  isModalOpen: boolean;
+};
+
+const initialState: DirtyFormStates = {
+  forms: [],
+  isModalOpen: false,
+};
+
+// Settings slice
+export const dirtyFormsSlice = createSlice({
```

Comment:
It'll be nice to have this available to all of our forms 🙂

Reply:
for sure!